agents/ten_packages/system/ten_ai_base/interface/ten_ai_base/llm.py

```diff
@@ -0,0 +1,119 @@
+#
+# This file is part of TEN Framework, an open source project.
+# Licensed under the Apache License, Version 2.0.
+# See the LICENSE file for more information.
+#
+from abc import ABC, abstractmethod
+import asyncio
+from enum import Enum
+from typing import Optional, TypedDict, Union
+
+from pydantic import BaseModel
+from ten import (
+    AsyncExtension,
+    TenEnv,
+    Data,
+)
+from ten.async_ten_env import AsyncTenEnv
+from ten.cmd import Cmd
+from ten.cmd_result import CmdResult, StatusCode
+from .const import CMD_PROPERTY_TOOL, CMD_TOOL_REGISTER
+from .types import TenLLMCallCompletionArgs, TenLLMDataCompletionArgs, TenLLMToolMetadata
+from .helper import AsyncQueue
+import json
+
+class TenLLMBaseExtension(AsyncExtension, ABC):
+    # Create the queue for message processing
+    queue = AsyncQueue()
+    available_tools: list[TenLLMToolMetadata] = []
+    available_tools_lock = asyncio.Lock()  # Lock to ensure thread-safe access
+    current_task = None
+
+    async def on_init(self, ten_env: TenEnv) -> None:
+        ten_env.log_debug("on_init")
+
+    async def on_start(self, ten_env: TenEnv) -> None:
+        ten_env.log_debug("on_start")
+
+        self.loop = asyncio.get_event_loop()
+        self.loop.create_task(self._process_queue(ten_env))
+
+    async def on_stop(self, ten_env: TenEnv) -> None:
+        ten_env.log_debug("on_stop")
+
+    async def on_deinit(self, ten_env: TenEnv) -> None:
+        ten_env.log_debug("on_deinit")
+
+    async def on_cmd(self, async_ten_env: AsyncTenEnv, cmd: Cmd) -> None:
```

Comment:
I think this method should return a bool to check whether the command is consumed by base class or not.

Reply:
this should really be done by ten runtime instead of us. if we are going to do this, we will need to define new lifecycle apis